### ISSUE TITLE
sqlite module with Node.js v6

### ISSUE THREAD

**Issue by M5oul** [duniter/duniter#381]
After `npm install` with success:
```fish
duniter -h
module.js:341
    throw err;
    ^

Error: Cannot find module '/root/duniter/node_modules/sqlite3b/lib/binding/node-v47-linux-arm/node_sqlite3.node'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/root/duniter/node_modules/sqlite3b/lib/sqlite3.js:4:15)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
```

**Comment by c-geek** [comment]
Nice catch. Could you remember me what is your hardware? (raspi, beaglebone, ...)

**Comment by M5oul** [comment]
Cubieboard.

**Comment by c-geek** [comment]
Ok now it compiles (alpha 38), but I get a an SQL error:

```
Error: SQLITE_ERROR: cannot start a transaction within a transaction
```

Will check that tomorrow.

**Comment by c-geek** [comment]
I could run duniter on ARM Raspberry PI Model B with 512MB RAM.

**Comment by M5oul** [comment]
I still have the issue as above.
UI install fail:
```fish
npm install
npm WARN lifecycle ucoin@0.20.0-a40~postinstall: cannot run in wd %s %s (wd=%s) ucoin@0.20.0-a40 mkdir ui -p && cd ui && npm pack ucoin-ui@0.1.18 && tar xzf ucoin-ui-0.1.18.tgz && cd package && npm install && cd .. && rm ucoin-ui-0.1.18.tgz /root/duniter
```

**Comment by c-geek** [comment]
You can try again :-)

**Comment by M5oul** [comment]
Still same issue.

**Comment by c-geek** [comment]
Tell me your opinion: what if I made an ARM .deb package instead? It would be installed as root as well, but at least I know it would work.

**Comment by M5oul** [comment]
Yep, a deb package would be great!

**Comment by M5oul** [comment]
I came to realize there is a [Duniter package for Debian](https://github.com/duniter/duniter/releases/download/v0.20.0a45/duniter-v0.20.0a45-linux-armv7l.deb) which works :smile_cat: 
I hope it was clear for you that I was talking about an installation from sources manually built.

**Comment by c-geek** [comment]
I don't get it, what do you mean? A .deb which manually build the sources? So the package here is not what you expected?

**Comment by M5oul** [comment]
This issue is about the fact that I encounter a bug using sources and not the deb package. I wanted to be sure that it's clear for you.
This package is very great :+1: Thanks.

**Comment by c-geek** [comment]
I made the .deb package for ARM because of this issue :-)

Yet, I need to install my Raspberry compilator to automatically upload the .deb on each release now: as you can see, only alpha45 is available!

**Comment by M5oul** [comment]
Ok, I better understand things.

**Comment by M5oul** [comment]
I encounter same issue on a Debian Jessie x86 in a Vagrant/Virtualbox:
```fish
vagrant@yunohost:~/duniter$ ./duniter.sh --version
module.js:442
    throw err;
    ^

Error: Cannot find module '/home/vagrant/duniter/node_modules/sqlite3b/lib/binding/Release/node-v48-linux-x64/node_sqlite3.node'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/vagrant/duniter/node_modules/sqlite3b/lib/sqlite3.js:4:15)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
```

I am using Node.js 6.2.0.

On the ARM board, there is Node.js v5.11.1.

**Comment by c-geek** [comment]
You need to make a `npm install` again with Node.js 6 before using commands.

**Comment by M5oul** [comment]
- [sqlite3.1.4 now support node v6](https://github.com/mapbox/node-sqlite3/issues/638)

- I found a [fix working](https://github.com/duniter/duniter/commit/1ee17a02d03216dec38c0ebc07807c273fc4e6b3).

Why do you we `sqlite3b`? I installed last version of [sqlite3.1.4](https://www.npmjs.com/package/sqlite3).

**Comment by c-geek** [comment]
Because at some point [it was not compliant with NW.js](https://github.com/c-geek/node-sqlite3/commit/d512b79ab3c16239cb8cc001328ccd24215cc358).

I can try again however, since we also upgraded NW.js version.

**Comment by c-geek** [comment]
Your commit may cause a crash during release building.. we will see.

**Comment by M5oul** [comment]
Fixed with https://github.com/duniter/duniter/pull/458.

**Comment by c-geek** [comment]
It has broken on Windows. Application is passing tests on AppVeyor, yet it fails to run after the installation. Some investigations tell me it is the sqlite3 package which do not success to load.

I will make a temporary `windows` branch to have a particular release for Windows with the old sqlite3 implementation.

**Comment by c-geek** [comment]
I've made a try, and everything seems to work: https://ci.appveyor.com/project/c-geek/duniter/history

I will make more tests tomorrow, but things are going very well for now 👍

**Comment by c-geek** [comment]
Fixed by #643
